File: leetcode/2288-apply-discount-to-prices/discountPrices_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type DiscountPricesCase struct {
	Input struct {
		Sentence string
		Discount int
	}

	Output string
}

func TestDiscountPrices(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]DiscountPricesCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := discountPrices(testCase.Input.Sentence, testCase.Input.Discount)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/2288-apply-discount-to-prices/discountPrices.test.ts
import cases from "./cases.json";
import discountPrices from "./discountPrices";

test("apply-discount-to-prices", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    console.time("time");
    const result = discountPrices(
      testCase.input.sentence,
      testCase.input.discount
    );
    console.timeEnd("time");
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});
